pytensor/link/pytorch/linker.py

```diff
@@ -25,7 +14,16 @@ def fgraph_convert(self, fgraph, input_storage, storage_map, **kwargs):
     def jit_compile(self, fn):
         import torch
 
-        return torch.compile(fn)
+        from pytensor.link.pytorch.dispatch import pytorch_typify
+
+        inner_fn = torch.compile(fn)
+
+        # Torch does not accept numpy inputs and may return GPU objects
+        def fn(*inputs, inner_fn=inner_fn):
+            outs = inner_fn(*(pytorch_typify(inp) for inp in inputs))
+            return tuple(out.cpu() for out in outs)
```

Comment:
I have a few PR's open that might cobble this fyi.

Reply:
Whichever gets merged first, we can solve the conflicts after